src/dmd/target.d

```diff
@@ -787,13 +795,39 @@ struct Target
      */
     extern (C++) static Expression getTargetInfo(const(char)* name, const ref Loc loc)
     {
-        switch (name.toDString)
-        {
-            case "cppRuntimeLibrary":
-                return new StringExp(loc, cast(void*)global.params.mscrtlib, global.params.mscrtlib ? strlen(global.params.mscrtlib) : 0);
+        int ival;
+        const(char)[] sval;
+        switch (name.toDString) with (TargetInfoKeys)
+        {
+            case objectFormat.stringof:
+                if (global.params.isWindows)
+                    sval = global.params.mscoff ? "coff" : "omf";
+                else if (global.params.isOSX)
+                    sval = "macho";
+                else
+                    sval = "elf";
+                goto LstringExpr;
+            case floatAbi.stringof:
+                sval = "hard";
+                goto LstringExpr;
+            case cppRuntimeLibrary.stringof:
+                if (global.params.isWindows)
+                {
+                    if (global.params.mscoff)
+                        sval = global.params.mscrtlib ? global.params.mscrtlib.toDString : "";
+                    else
+                        sval = "snn";
+                }
+                else
+                    sval = "";
+                goto LstringExpr;
             default:
                 return null;
         }
+    LintExpr:
+        return new IntegerExp(loc, ival, Type.tint32);
+    LstringExpr:
+        return new StringExp(loc, cast(void*)sval.ptr, sval.length);
```

Comment:
Replace gotos with inner functions. See for example: https://github.com/dlang/dmd/pull/6656/files

Reply:
I was inspired by code literally all over DMD to use that style...

Reply:
I guessed so, but we want to remove bad practices from the code base, see also: https://forum.dlang.org/thread/p6oibo$1lmi$1@digitalmars.com

Reply:
Incidentally, I think the local functions in your example are possibly uglier than the code they replace (some goto labels)... but they're certainly more 'principled' ;)